src/server/pachyderm_test.go

```diff
@@ -1146,85 +1146,130 @@ func TestPipelineErrorHandling(t *testing.T) {
-		&pps.CreatePipelineRequest{
-			Pipeline: client.NewPipeline(pipeline),
-			Transform: &pps.Transform{
-				Cmd:      []string{"bash"},
-				Stdin:    []string{"if", fmt.Sprintf("[ -a pfs/%v/file1 ]", dataRepo), "then", "exit 0", "fi", "exit 1"},
-				ErrCmd:   []string{"bash"},
-				ErrStdin: []string{"if", fmt.Sprintf("[ -a pfs/%v/file3 ]", dataRepo), "then", "exit 0", "fi", "exit 1"},
-			},
-			Input: client.NewPFSInput(dataRepo, "/*"),
-		})
-	require.NoError(t, err)
+		_, err := c.PutFile(dataRepo, "master", "file1", strings.NewReader("foo\n"))
+		require.NoError(t, err)
+		_, err = c.PutFile(dataRepo, "master", "file2", strings.NewReader("bar\n"))
+		require.NoError(t, err)
+		_, err = c.PutFile(dataRepo, "master", "file3", strings.NewReader("bar\n"))
+		require.NoError(t, err)
 
-	var jobInfos []*pps.JobInfo
-	require.NoError(t, backoff.Retry(func() error {
-		jobInfos, err = c.ListJob(pipeline, nil, nil, -1, true)
+		// In this pipeline, we'll have a command that fails for files 2 and 3, and an error handler that fails for file 2
+		pipeline := tu.UniqueString("pipeline1")
+		_, err = c.PpsAPIClient.CreatePipeline(
+			context.Background(),
+			&pps.CreatePipelineRequest{
+				Pipeline: client.NewPipeline(pipeline),
+				Transform: &pps.Transform{
+					Cmd:      []string{"bash"},
+					Stdin:    []string{"if", fmt.Sprintf("[ -a pfs/%v/file1 ]", dataRepo), "then", "exit 0", "fi", "exit 1"},
+					ErrCmd:   []string{"bash"},
+					ErrStdin: []string{"if", fmt.Sprintf("[ -a pfs/%v/file3 ]", dataRepo), "then", "exit 0", "fi", "exit 1"},
+				},
+				Input: client.NewPFSInput(dataRepo, "/*"),
+			})
 		require.NoError(t, err)
-		if len(jobInfos) != 1 {
-			return fmt.Errorf("expected 1 job, got %d", len(jobInfos))
-		}
-		return nil
-	}, backoff.NewTestingBackOff()))
-	jobInfo, err := c.PpsAPIClient.InspectJob(context.Background(), &pps.InspectJobRequest{
-		Job:        jobInfos[0].Job,
-		BlockState: true,
+
+		jis, err := c.FlushJobAll([]*pfs.Commit{client.NewCommit(dataRepo, "master")}, nil)
+		require.NoError(t, err)
+		require.Equal(t, 1, len(jis))
+		jobInfo := jis[0]
+
+		// We expect the job to fail, and have 1 datum processed, recovered, and failed each
+		require.Equal(t, pps.JobState_JOB_FAILURE, jobInfo.State)
+		require.Equal(t, int64(1), jobInfo.DataProcessed)
+		require.Equal(t, int64(1), jobInfo.DataRecovered)
+		require.Equal(t, int64(1), jobInfo.DataFailed)
+
+		// For this pipeline, we have the same command as before, but this time the error handling passes for all
+		pipeline = tu.UniqueString("pipeline2")
+		_, err = c.PpsAPIClient.CreatePipeline(
+			context.Background(),
+			&pps.CreatePipelineRequest{
+				Pipeline: client.NewPipeline(pipeline),
+				Transform: &pps.Transform{
+					Cmd:    []string{"bash"},
+					Stdin:  []string{"if", fmt.Sprintf("[ -a pfs/%v/file1 ]", dataRepo), "then", "exit 0", "fi", "exit 1"},
+					ErrCmd: []string{"true"},
+				},
+				Input: client.NewPFSInput(dataRepo, "/*"),
+			})
```

Comment:
Could we do this as an update to the previous pipeline instead of creating a new pipeline? That way we can check if the datum that processed successfully above is actually getting skipped in this version of the pipeline.